### ISSUE TITLE
Custom LifecycleHook Pods not deleted & Task phase not updated 

### ISSUE THREAD

**Issue by Piroddi** [argoproj/argo-workflows#8293]
## Checklist

 

* [x] Double-checked my configuration.
* [x] Tested using the latest version.
* [x] Used the Emissary executor.

## Summary

Tasks phases from Custom LifecycleHook's at a workflow level are not being updated after completion. This results in the phase being stuck in 'Pending' & the hook pod not being deleted when the workflow has the following spec 

```yaml
  podGC:
    strategy: OnPodCompletion   
```

<img width="1232" alt="image" src="https://user-images.githubusercontent.com/24908789/161249610-8b6bb871-dfd9-4629-ad70-264f5d7edfd6.png">

<img width="859" alt="image" src="https://user-images.githubusercontent.com/24908789/161251657-84790de2-3520-4133-8414-5577fce92782.png">

Workflow details:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  annotations:
    workflows.argoproj.io/pod-name-format: v1
  creationTimestamp: "2022-04-01T10:27:26Z"
  generateName: lifecycle-hook-tmpl-level-
  generation: 4
  labels:
    workflows.argoproj.io/completed: "true"
    workflows.argoproj.io/phase: Failed
    workflows.argoproj.io/resubmitted-from-workflow: lifecycle-hook-tmpl-level-nddnn
    workflows.argoproj.io/workflow-archiving-status: Archived
  name: lifecycle-hook-tmpl-level-dq5bd
  namespace: argo
  resourceVersion: "334749933"
  uid: 01acc8ce-7075-48e1-b0a1-3995bcd18a94
spec:
  arguments: {}
  entrypoint: main
  hooks:
    failed:
      arguments: {}
      expression: workflow.status == "Failed"
      template: echo-save
  nodeSelector:
    nodeGroup: argo
  podGC:
    strategy: OnPodCompletion
  serviceAccountName: argo-workflows
  templates:
  - inputs: {}
    metadata: {}
    name: main
    outputs: {}
    steps:
    - - arguments: {}
        name: step-1
        template: echo-save
    - - arguments: {}
        name: step2
        template: echo
  - container:
      args:
      - echoss "it was heads"
      command:
      - sh
      - -c
      image: alpine:3.6
      name: ""
      resources: {}
    inputs: {}
    metadata: {}
    name: echo
    outputs: {}
  - container:
      args:
      - echo "it was heads"
      command:
      - sh
      - -c
      image: alpine:3.6
      name: ""
      resources: {}
    inputs: {}
    metadata: {}
    name: echo-save
    outputs: {}
  - http:
      url: https://raw.githubusercontent.com/argoproj/argo-workflows/4e450e250168e6b4d51a126b784e90b11a0162bc/pkg/apis/workflow/v1alpha1/generated.swagger.json
    inputs: {}
    metadata: {}
    name: http
    outputs: {}
  tolerations:
  - effect: NoSchedule
    key: nodeGroup
    value: argo
status:
  artifactRepositoryRef:
    artifactRepository:
      *
    default: true
  conditions:
  - status: "False"
    type: PodRunning
  - status: "True"
    type: Completed
  finishedAt: "2022-04-01T10:27:46Z"
  message: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed
  nodes:
    lifecycle-hook-tmpl-level-dq5bd:
      children:
      - lifecycle-hook-tmpl-level-dq5bd-2217863861
      - lifecycle-hook-tmpl-level-dq5bd-3701634578
      displayName: lifecycle-hook-tmpl-level-dq5bd
      finishedAt: "2022-04-01T10:27:46Z"
      id: lifecycle-hook-tmpl-level-dq5bd
      message: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed
      name: lifecycle-hook-tmpl-level-dq5bd
      outboundNodes:
      - lifecycle-hook-tmpl-level-dq5bd-2503513326
      phase: Failed
      progress: 1/3
      resourcesDuration:
        cpu: 5
        memory: 5
      startedAt: "2022-04-01T10:27:26Z"
      templateName: main
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: Steps
    lifecycle-hook-tmpl-level-dq5bd-798385017:
      boundaryID: lifecycle-hook-tmpl-level-dq5bd
      children:
      - lifecycle-hook-tmpl-level-dq5bd-2150900480
      displayName: step-1
      finishedAt: "2022-04-01T10:27:30Z"
      hostNodeName: *
      id: lifecycle-hook-tmpl-level-dq5bd-798385017
      name: lifecycle-hook-tmpl-level-dq5bd[0[].step-1
      outputs:
        artifacts:
        - name: main-logs
          s3:
            key: *
        exitCode: "0"
      phase: Succeeded
      progress: 1/1
      resourcesDuration:
        cpu: 3
        memory: 3
      startedAt: "2022-04-01T10:27:26Z"
      templateName: echo-save
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: Pod
    lifecycle-hook-tmpl-level-dq5bd-2150900480:
      boundaryID: lifecycle-hook-tmpl-level-dq5bd
      children:
      - lifecycle-hook-tmpl-level-dq5bd-2503513326
      displayName: '[1[]'
      finishedAt: "2022-04-01T10:27:46Z"
      id: lifecycle-hook-tmpl-level-dq5bd-2150900480
      message: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed
      name: lifecycle-hook-tmpl-level-dq5bd[1[]
      phase: Failed
      progress: 0/1
      resourcesDuration:
        cpu: 2
        memory: 2
      startedAt: "2022-04-01T10:27:36Z"
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: StepGroup
    lifecycle-hook-tmpl-level-dq5bd-2217863861:
      boundaryID: lifecycle-hook-tmpl-level-dq5bd
      children:
      - lifecycle-hook-tmpl-level-dq5bd-798385017
      displayName: '[0[]'
      finishedAt: "2022-04-01T10:27:36Z"
      id: lifecycle-hook-tmpl-level-dq5bd-2217863861
      name: lifecycle-hook-tmpl-level-dq5bd[0[]
      phase: Succeeded
      progress: 1/2
      resourcesDuration:
        cpu: 5
        memory: 5
      startedAt: "2022-04-01T10:27:26Z"
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: StepGroup
    lifecycle-hook-tmpl-level-dq5bd-2503513326:
      boundaryID: lifecycle-hook-tmpl-level-dq5bd
      displayName: step2
      finishedAt: "2022-04-01T10:27:40Z"
      hostNodeName:*
      id: lifecycle-hook-tmpl-level-dq5bd-2503513326
      message: Error (exit code 127)
      name: lifecycle-hook-tmpl-level-dq5bd[1[].step2
      outputs:
        artifacts:
        - name: main-logs
          s3:
            key: *
        exitCode: "127"
      phase: Failed
      progress: 0/1
      resourcesDuration:
        cpu: 2
        memory: 2
      startedAt: "2022-04-01T10:27:36Z"
      templateName: echo
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: Pod
    lifecycle-hook-tmpl-level-dq5bd-3701634578:
      displayName: lifecycle-hook-tmpl-level-dq5bd.hooks.failed
      finishedAt: null
      id: lifecycle-hook-tmpl-level-dq5bd-3701634578
      name: lifecycle-hook-tmpl-level-dq5bd.hooks.failed
      phase: Pending
      progress: 0/1
      startedAt: "2022-04-01T10:27:46Z"
      templateName: echo-save
      templateScope: local/lifecycle-hook-tmpl-level-dq5bd
      type: Pod
  phase: Failed
  progress: 1/3
  resourcesDuration:
    cpu: 5
    memory: 5
  startedAt: "2022-04-01T10:27:26Z"
```

What happened/what you expected to happen?

The phase of a custom hooks should be updated after completion & pod deleted 

What version are you running?

latest

## Diagnostics

# Logs from the workflow controller:
```
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.545Z" level=info msg="Processing workflow" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=1 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="task-result changed" namespace=argo nodeID=lifecycle-hook-tmpl-level-dq5bd-798385017 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="node changed" new.message= new.phase=Succeeded new.progress=0/1 nodeID=lifecycle-hook-tmpl-level-dq5bd-798385017 old.message= old.phase=Pending old.progress=0/1
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="Step group node lifecycle-hook-tmpl-level-dq5bd-2217863861 successful" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd-2217863861 phase Running -> Succeeded" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd-2217863861 finished: 2022-04-01 10:27:36.546778693 +0000 UTC" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="StepGroup node lifecycle-hook-tmpl-level-dq5bd-2150900480 initialized Running" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="SG Outbound nodes of lifecycle-hook-tmpl-level-dq5bd-798385017 are [lifecycle-hook-tmpl-level-dq5bd-798385017]" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.546Z" level=info msg="Pod node lifecycle-hook-tmpl-level-dq5bd-2503513326 initialized Pending" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.592Z" level=info msg="Create pods 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.594Z" level=info msg="Created pod: lifecycle-hook-tmpl-level-dq5bd[1].step2 (lifecycle-hook-tmpl-level-dq5bd-2503513326)" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.594Z" level=info msg="Workflow step group node lifecycle-hook-tmpl-level-dq5bd-2150900480 not yet completed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.594Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.594Z" level=info msg=reconcileAgentPod namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.613Z" level=info msg="Update workflows 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.613Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=334749848 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.622Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.633Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.640Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:36.649Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:41.117Z" level=info msg="Get leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:41.126Z" level=info msg="Update leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:41.614Z" level=info msg="cleaning up pod" action=deletePod key=argo/lifecycle-hook-tmpl-level-dq5bd-798385017/deletePod
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:41.640Z" level=info msg="Delete pods 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.144Z" level=info msg="Get leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.158Z" level=info msg="Update leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.601Z" level=info msg="Processing workflow" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.602Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=2 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.602Z" level=info msg="task-result changed" namespace=argo nodeID=lifecycle-hook-tmpl-level-dq5bd-2503513326 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.602Z" level=info msg="node changed" new.message="Error (exit code 127)" new.phase=Failed new.progress=0/1 nodeID=lifecycle-hook-tmpl-level-dq5bd-2503513326 old.message= old.phase=Pending old.progress=0/1
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="SG Outbound nodes of lifecycle-hook-tmpl-level-dq5bd-798385017 are [lifecycle-hook-tmpl-level-dq5bd-798385017]" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="Step group node lifecycle-hook-tmpl-level-dq5bd-2150900480 deemed failed: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd-2150900480 phase Running -> Failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd-2150900480 message: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd-2150900480 finished: 2022-04-01 10:27:46.603213243 +0000 UTC" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="step group lifecycle-hook-tmpl-level-dq5bd-2150900480 was unsuccessful: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="Outbound nodes of lifecycle-hook-tmpl-level-dq5bd-2503513326 is [lifecycle-hook-tmpl-level-dq5bd-2503513326]" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="Outbound nodes of lifecycle-hook-tmpl-level-dq5bd is [lifecycle-hook-tmpl-level-dq5bd-2503513326]" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd phase Running -> Failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd message: child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="node lifecycle-hook-tmpl-level-dq5bd finished: 2022-04-01 10:27:46.603313974 +0000 UTC" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.603Z" level=info msg="Checking daemoned children of lifecycle-hook-tmpl-level-dq5bd" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.612Z" level=info msg="Running workflow level hooks" lifeCycleHook=failed namespace=argo node=lifecycle-hook-tmpl-level-dq5bd.hooks.failed workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.612Z" level=info msg="Pod node lifecycle-hook-tmpl-level-dq5bd-3701634578 initialized Pending" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.643Z" level=info msg="Create pods 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Created pod: lifecycle-hook-tmpl-level-dq5bd.hooks.failed (lifecycle-hook-tmpl-level-dq5bd-3701634578)" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg=reconcileAgentPod namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Updated phase Running -> Failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Updated message  -> child 'lifecycle-hook-tmpl-level-dq5bd-2503513326' failed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Marking workflow completed" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Marking workflow as pending archiving" namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.650Z" level=info msg="Checking daemoned children of " namespace=argo workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.655Z" level=info msg="cleaning up pod" action=deletePod key=argo/lifecycle-hook-tmpl-level-dq5bd-1340600742-agent/deletePod
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.659Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.661Z" level=info msg="Delete pods 404"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.664Z" level=info msg="Update workflows 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.665Z" level=info msg="Workflow update successful" namespace=argo phase=Failed resourceVersion=334749923 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.673Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.683Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.692Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.712Z" level=info msg="Create events 201"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.713Z" level=info msg="DeleteCollection workflowtaskresults 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.714Z" level=info msg="archiving workflow" namespace=argo uid=01acc8ce-7075-48e1-b0a1-3995bcd18a94 workflow=lifecycle-hook-tmpl-level-dq5bd
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:46.812Z" level=info msg="Patch workflows 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:51.172Z" level=info msg="Get leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:51.181Z" level=info msg="Update leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:51.714Z" level=info msg="cleaning up pod" action=deletePod key=argo/lifecycle-hook-tmpl-level-dq5bd-2503513326/deletePod
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:51.738Z" level=info msg="Delete pods 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:56.199Z" level=info msg="Get leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:27:56.208Z" level=info msg="Update leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:28:01.215Z" level=info msg="Get leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:28:01.224Z" level=info msg="Update leases 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:28:02.442Z" level=info msg="List workflows 200"
workflow-controller-6b6746fbb6-5qxt4 time="2022-04-01T10:28:02.442Z" level=info msg=healthz age=5m0s err="<nil>" instanceID= labelSelector="!workflows.argoproj.io/phase,!workflows.argoproj.io/controller-instanceid" managedNamespace=
```
---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by Piroddi** [comment]
Hi @sarabala1979 could you please confirm if this is a bug and being worked on? 

Can then remove stale label

**Comment by Piroddi** [comment]
@sarabala1979 the podGC issue was resolved with #8452. 

However lifecycle hooks run from a workflow level, get stuck in 'Pending', even after they have completed. 

Should I close this issues and create a new one specific to the 'Pending status'. Its currently causing issues when trying to retry a failed workflow. 

![Screenshot 2022-04-27 at 13 48 41](https://user-images.githubusercontent.com/24908789/165511909-56ede296-74cc-413b-983c-9c6a177feba0.png)

**Comment by Piroddi** [comment]
PodGC resolved by #8448. However hook phased not updated. Created a more specific issue: #8529
